wp-content/themes/sample-theme/blocks/src/core--button/block.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace XWP\Sample_Theme\Blocks;
+
+use XWP\Sample_Theme\Block;
+use XWP\Sample_Theme\Block_Script;
+
+$block = new Block( 'xwp-sample-theme/button' );
+
+$block->extends( 'core/button' );
+
+$index_script = new Block_Script( __DIR__ . '/script.js' );
+$block->add_asset( 'script', $index_script->script() );
+$block->add_asset( 'style', $index_script->style() );
```

Comment:
This helper allows us to reference the actual `wp-scripts build` entrypoint filenames and it knows how they'll be renamed after the build.

Contrast this to `block.json` props for `script`, `style`, etc. which require you to specify the _built_ asset path.